### ISSUE TITLE
Please provide a binary for aarch64 + deb10

### ISSUE THREAD

**Issue by tarleb** [haskell/cabal#8849]
It seems that the [list of official binaries](https://downloads.haskell.org/~cabal/cabal-install-3.10.1.0/) does currently not include a build for aarch64 on deb10. Adding this build would allow us to publish a new Docker image with GHC 9.6 and cabal-install 3.10.

See also: haskell/docker-haskell#102

**Comment by Mikolaj** [comment]
Hi! The cabal team is not able to produce these binaries, but @hasufell over at ghcup has built them (if these are what you need): https://github.com/haskell/ghcup-metadata/commit/b703c559915440050b9083983b6e49262761e654

**Comment by Mikolaj** [comment]
BTW, is a GHC binary for exactly this platform produced by gitlab CI? If so, could you point me to a gitlab pipeline doing exactly that?

**Comment by hasufell** [comment]
> The cabal team is not able to produce these binaries

Why?

**Comment by tarleb** [comment]
Official Docker images cannot contain unofficial binaries afaik, so we are somewhat dependent on the cabal project to provide those. How can we help to produce those, and where would we start?

**Comment by hasufell** [comment]
It appears this was a commit by @chreekat removing aarch64 linux: https://github.com/haskell/cabal/commit/0ea1001301427a0911f38679763f7617acef28f7

The rationale was

> GHCUp fails to run on aarch64-linux-deb10 due to a glibc mismatch.
> 
> aarch64-linux-deb11 would work, but there is no GHC release for that
version yet.

At the time of the commit, this was already fixed in GHCup. See the changelog: https://github.com/haskell/ghcup-hs/blob/master/CHANGELOG.md#01191----2023-2-19

And the ticket: https://github.com/haskell/ghcup-hs/issues/762

Further, the second part of the rationale doesn't make sense to me. You should be able to install the deb10 GHC on deb11.

It seems I wasn't consulted on the PR either.

**Comment by Kleidukos** [comment]
Looks like the path to resolution is clearer now.

**Comment by chreekat** [comment]
> It seems I wasn't consulted on the PR either.

My mistake, that was an oversight. I was just trying to get CI to green and forgot to double back on some of the issues I ran into.

I'm surprised to hear it was already fixed in GHCUp, though. The CI script just basically runs `curl  https://get-ghcup.haskell.org | sh` — does it need to do something different?

**Comment by Kleidukos** [comment]
@hasufell @chreekat Any chance we could sync on this btw? CI is highly critical for cabal and I need to make sure that institutional knowledge gets written and understood by other maintainers.

**Comment by angerman** [comment]
Can I bring up again, that we should have a mailinglist for these kinds of issues?

Currently you have to sift through various GitHub notifications, potentially some IRC/Matrix/Telegram chats... we should really have a mailing list with the interested CI related parties involved to keep everyone up to date what's happening across the whole GitHub:haskell ecosystem.

**Comment by Kleidukos** [comment]
Yes there is considerable overhead.

**Comment by hasufell** [comment]
> I'm surprised to hear it was already fixed in GHCUp, though. The CI script just basically runs curl https://get-ghcup.haskell.org | sh — does it need to do something different?

I'm surprised too.

The bootstrap script was updated in time: https://github.com/haskell/ghcup-hs/commit/ea6c8d338c41ca6ae020685cd3eab263206218e0

The ghcup release CI shows that deb10 was used to build: https://github.com/haskell/ghcup-hs/actions/runs/4254450877/jobs/7400717833

GHCup knows only Tier 1 platforms. So everything like that is a bug that I'll fix next weekend.

**Comment by chreekat** [comment]
Maybe I should run an apt-get update or something?

I re-enabled aarch64-deb10 and aarch64-deb11 jobs, and sure enough the same thing happens:

https://gitlab.haskell.org/haskell/cabal/-/jobs/1404272

**Comment by chreekat** [comment]
deb11 seems to be faring better — it's at least compiling Cabal now. https://gitlab.haskell.org/haskell/cabal/-/jobs/1404273.

I hadn't considered that "there is no GHC release for aarch64-deb11" wouldn't be a blocker.

**Comment by chreekat** [comment]
Doing an update && upgrade with apt-get doesn't help.

Installing the [system requirements](https://www.haskell.org/ghcup/install/#system-requirements) doesn't help, either (although it *does* install missing packages...)

Unrelated:  in Debian 11, installing the listed system requirements fails with `E: Unable to locate package libffi6`

**Comment by chreekat** [comment]
> Can I bring up again, that we should have a mailinglist for these kinds of issues?

Almost missed this message. I will see about getting a mailing list created. (There is an existing ghc-devops group, but its charter was explicitly "only ghc", which is not what we want.)

**Comment by Kleidukos** [comment]
Yes the GHC DevOps Group was an endearing experiment but clearly we need to think our operations and infrastructure at the scale of the Haskell project now.

**Comment by hasufell** [comment]
@chreekat I found the problem and will fix it tonight

**Comment by hasufell** [comment]
https://gitlab.haskell.org/haskell/cabal/-/jobs/1404425

**Comment by chreekat** [comment]
https://github.com/haskell/cabal/pull/8851

**Comment by chreekat** [comment]
aarch64-linux-deb10 is back on the menu, e.g. https://downloads.haskell.org/cabal/cabal-install-3.10.2.0/cabal-install-3.10.2.0-aarch64-linux-deb10.tar.xz. Anything else we should do as followup?

Closing in the meanwhile.
